proposals/lip-0053.md

```diff
@@ -601,16 +598,16 @@ def verify(trs: Transaction) -> None:
 
     if len(certificateBytes) > 0:
         verifyCertificate(ccu)
-    
-    # Liveness condition is only checked on the mainchain for the first CCU with a non-empty inbox update.
-    if chainAccount(ccu.params.sendingChainID).status == CHAIN_STATUS_REGISTERED and ccu.params.inboxUpdate is not empty:
-        verifyLivenessConditionForRegisteredChains(ccu)
 
     validatorsUpdate = ccu.params.activeValidatorsUpdate
     if len(validatorsUpdate.blsKeysUpdate) > 0 or len(validatorsUpdate.bftWeightsUpdate) > 0 or validatorsUpdate.bftWeightsUpdateBitmap != EMPTY_BYTES or ccu.params.certificateThreshold != validators(ccu.params.sendingChainID).certificateThreshold:
         verifyValidatorsUpdate(ccu)
     if ccu.params.inboxUpdate is not empty:
         verifyPartnerChainOutboxRoot(ccu)
+
+    # Liveness condition is only checked on the mainchain for the first CCU with a non-empty inbox update.
+    if ownChainAccount.chainID == getMainchainID() and chainAccount(ccu.params.sendingChainID).status == CHAIN_STATUS_REGISTERED and ccu.params.inboxUpdate is not empty:
```

Comment:
It's probably quite confusing for some readers why we have the check `if ownChainAccount.chainID == getMainchainID()` inthe  verify function for the mainchain. Therefore, it might be better to move the logic to an internal function which is then called from both `verify` functions.

Reply:
done